tlatools/org.lamport.tlatools/test/tlc2/debug/GetScopedIdentifiersTests.java

```diff
@@ -0,0 +1,134 @@
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN
+ * AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ ******************************************************************************/
+
+package tlc2.debug;
+
+import util.ParserAPI;
+import util.UniqueString;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+import org.junit.Assert;
+
+import tla2sany.semantic.ModuleNode;
+import tla2sany.st.Location;
+
+/**
+ * {@link TLCBreakpointExpression#getScopedIdentifiers(ModuleNode, Location)}
+ * tests, to check that various nested TLA+ syntax constructs have their
+ * introduced identifiers appropriately recorded.
+ */
+@RunWith(Parameterized.class)
+public class GetScopedIdentifiersTests {
+
+  private static class TestCase {
+    public final String input;
+    public final String terminator;
+    public final Set<String> expected;
+    public TestCase(String input, String terminator, String... expected) {
+      this.input = input;
+      this.terminator = terminator;
+      this.expected = new HashSet<String>(List.of(expected));
+    }
+    public String toString() {
+    	return this.input + " TRUE " + this.terminator;
+    }
+  }
+
+  /**
+   * A wrapper in which to interpolate test cases to form a valid module.
+   */
+  private static final String wrapper =
+    "---- MODULE Test ----\n"
+    + "VARIABLE x\n"
+    + "CONSTANT y\n"
+    + "%s\n"
+    + "TRUE%s\n"
+    + "====";
+
+  /**
+   * The location of the expression TRUE in the wrapper. This assumes that
+   * each test case input is a single line.
+   */
+  private static final Location location
+    = new Location(UniqueString.of("Test"), 5, 1, 5, 2);
+
+  /**
+   * A set of test cases for finding scoped identifiers. Each input should
+   * consist of an incomplete operator definition & expression that can be
+   * terminated by expression TRUE located at the start of the next line,
+   * along with a variable-length list of identifiers accessible at the
+   * point of the TRUE expression. A terminator to be appended after TRUE is
+   * also accepted - for example ⟩ or ) or ] or }.
+   */
+  @Parameters(name = "{index}: {0}")
+  public static TestCase[] testCases() {
+    return new TestCase[] {
+      new TestCase("op ≜", ""),
+      new TestCase("op(i, j, k) ≜", "", "j", "i", "k"),
```

Comment:
This doesn't have to be changed, but in general, I want to use ASCII in our test suite to maintain compatibility with older versions of tla2tools that were released before the Unicode feature was introduced. This ensures we can run new tests on those older versions should this become necessary.